### ISSUE TITLE
Bash on Ubuntu on Windows - Internet connection error

### ISSUE THREAD

**Issue by j86989** [tldr-pages/tldr-python-client#56]
## Reproduction Rate
I could reproduce the issue each time I tried to print any `TLDR` page.

## Steps to Reproduce
1. Install Windows Subsystem for Linux.
2. Install Ubuntu on Windows.
3. Run Bash on Ubuntu on Windows.
4. Install `linuxbrew`. There is installation instruction on `http://linuxbrew.sh`.
5. Install `Python3`: `brew update` and `brew install python3`.
6. Install `TLDR` `Python` client: `pip3 install tldr`.
6. Try to print any `TLDR` page.

## Result
```bash
$ tldr tldr
No internet connection detected. Please reconnect and try again.
```
```bash
$ tldr apt
No internet connection detected. Please reconnect and try again.
```

## Expected Result
`TLDR` pages

## Additional Information
Windows Subsystem for Linux, Bash on Ubuntu on Windows

I have reproduced the issue on `AIX 7.1`, `Python 2.7.12`.

**Comment by JacobCallahan** [comment]
are you able to make any outbound connections with requests?

**Comment by j86989** [comment]
I'm not sure. I'm behind firewall.

```
$ ping -c 6 github.com

PING github.com (192.30.253.112) 56(84) bytes of data.
64 bytes from lb-192-30-253-112-iad.github.com (192.30.253.112): icmp_seq=1 ttl=48 time=111 ms
64 bytes from lb-192-30-253-112-iad.github.com (192.30.253.112): icmp_seq=2 ttl=48 time=113 ms
64 bytes from lb-192-30-253-112-iad.github.com (192.30.253.112): icmp_seq=3 ttl=48 time=137 ms
64 bytes from lb-192-30-253-112-iad.github.com (192.30.253.112): icmp_seq=4 ttl=48 time=113 ms
64 bytes from lb-192-30-253-112-iad.github.com (192.30.253.112): icmp_seq=5 ttl=48 time=112 ms
64 bytes from lb-192-30-253-112-iad.github.com (192.30.253.112): icmp_seq=6 ttl=48 time=112 ms

--- github.com ping statistics ---
6 packets transmitted, 6 received, 0% packet loss, time 5004ms
rtt min/avg/max/mdev = 111.971/116.772/137.721/9.401 ms
```

**Comment by JacobCallahan** [comment]
try
```python
import requests
result = requests.get('http://google.com')
print(result.status_code)
```
if the connection is good, you should see ```200```

**Comment by j86989** [comment]
```
$ python3
Python 3.6.4 (default, Feb 26 2018, 11:13:22)
[GCC 5.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> print(requests.get('http://google.com'))
<Response [200]>
```

**Comment by denbkh** [comment]
I have the same error with Cygwin on Windows. The interesting thing is that some commands work and some don't. 
For example: 
```
$ tldr dpkg 
No internet connection detected. Please reconnect and try again.
```

```
$ tldr tar 
# tar

  Archiving utility.
  Often combined with a compression method, such as gzip or bzip.
...
```
Maybe this is because "tar" is placed in "pages/common" but dpkg is in "pages/linux" ?

**Comment by TimothyBramlett** [comment]
Same problem on OS X behind a corporate proxy. Terminal Session definitely has internet.

and `import requests
result = requests.get('http://google.com')
print(result.status_code)` results in 200

**Comment by MasterOdin** [comment]
It's probably more important to test that you can access a page within tldr-pages and not just google:
```
>>> import requests
>>> r = requests.get('https://raw.github.com/tldr-pages/tldr/master/pages/common/tar.md')
<Response [200]>
>>> print(r.status_code)
>>> print(r.text)
```

**Comment by TimothyBramlett** [comment]
@MasterOdin Would it not be: `https://raw.githubusercontent.com/tldr-pages/tldr/master/pages/common/tar.md` ?  I guess its supposed to auto redirect.  But your assertion seems to be right and I think the proxy is blocking me.

**Comment by MasterOdin** [comment]
So doing:
```
>>> import requests
>>> requests.get('https://raw.github.com/tldr-pages/tldr/master/pages/common/tar.md')
```

fails while doing:
```
>>> import requests
>>> requests.get('https://raw.githubusercontent.com/tldr-pages/tldr/master/pages/common/tar.md')
```
works?

I've opened #67 to use the new link as that's apparently the recommended thing to do since 2014 (https://developer.github.com/changes/2014-04-25-user-content-security/)

**Comment by TimothyBramlett** [comment]
@MasterOdin Exactly.  But I think that is because my proxy is not filtering `raw.githubusercontent.com`.   But yeah it might be good to use the new link since the old link `raw.github.com` supposedly doesn't always redirect to `raw.githubusercontent.com`  https://github.com/processing/processing-docs/issues/532

**Comment by xmfbit** [comment]
Changing `urlopen` to `requests` solved my problem. You can find  the  
executable file `tldr` in `/path/of/anaconda/bin` if you are using Anaconda. Change the line https://github.com/tldr-pages/tldr-python-client/blob/master/tldr.py#L141 to
```
data = requests.get(page_url).text
```

**Comment by j86989** [comment]
```
Python 3.7.0 (default, Jul  1 2018, 10:02:32)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> requests.get('https://raw.github.com/tldr-pages/tldr/master/pages/common/tar.md')
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connectionpool.py", line 343, in _make_request
    self._validate_conn(conn)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connectionpool.py", line 849, in _validate_conn
    conn.connect()
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connection.py", line 356, in connect
    ssl_context=context)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/util/ssl_.py", line 359, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/ssl.py", line 412, in wrap_socket
    session=session
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/ssl.py", line 850, in _create
    self.do_handshake()
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/ssl.py", line 1108, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1045)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/adapters.py", line 445, in send
    timeout=timeout
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/util/retry.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /tldr-pages/tldr/master/pages/common/tar.md (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1045)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/sessions.py", line 644, in send
    history = [resp for resp in gen] if allow_redirects else []
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/sessions.py", line 644, in <listcomp>
    history = [resp for resp in gen] if allow_redirects else []
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/sessions.py", line 222, in resolve_redirects
    **adapter_kwargs
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/adapters.py", line 511, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /tldr-pages/tldr/master/pages/common/tar.md (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1045)')))
>>> requests.get('https://raw.githubusercontent.com/tldr-pages/tldr/master/pages/common/tar.md')
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connectionpool.py", line 343, in _make_request
    self._validate_conn(conn)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connectionpool.py", line 849, in _validate_conn
    conn.connect()
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connection.py", line 356, in connect
    ssl_context=context)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/util/ssl_.py", line 359, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/ssl.py", line 412, in wrap_socket
    session=session
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/ssl.py", line 850, in _create
    self.do_handshake()
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/ssl.py", line 1108, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1045)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/adapters.py", line 445, in send
    timeout=timeout
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/urllib3/util/retry.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /tldr-pages/tldr/master/pages/common/tar.md (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1045)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/python/3.7.0/lib/python3.7/site-packages/requests/adapters.py", line 511, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /tldr-pages/tldr/master/pages/common/tar.md (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1045)')))
>>>
```

**Comment by j86989** [comment]
```
Python 2.7.12 (default, Feb 20 2017, 03:58:51) [C] on aix6
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> requests.get('https://raw.github.com/tldr-pages/tldr/master/pages/common/tar.md')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/adapters.py", line 495, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', error(73, 'Connection reset by peer'))
>>> requests.get('https://raw.githubusercontent.com/tldr-pages/tldr/master/pages/common/tar.md')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "/app/abiniszio-zw/python/opt/freeware/lib/requests/adapters.py", line 511, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /tldr-pages/tldr/master/pages/common/tar.md (Caused by SSLError(SSLError(1, u'[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)'),))
>>> 
```

**Comment by evertvr** [comment]
> 
> 
> Changing `urlopen` to `requests` solved my problem. You can find the
> executable file `tldr` in `/path/of/anaconda/bin` if you are using Anaconda. Change the line https://github.com/tldr-pages/tldr-python-client/blob/master/tldr.py#L141 to
> 
> ```
> data = requests.get(page_url).text
> ```
> 
> And don't forget to import requests at the beginning of the file.

I've tried this, but it's not working for me, still gives "No internet connection detected. Please reconnect and try again."
I'm on Windows 10 WSL

**Comment by zdroid** [comment]
Can you re-try this with the latest version of `tldr.py` (from master)? It may have a fix for what was issue here.

**Comment by zdroid** [comment]
**UPDATE**:

This should be fixed in #39. Several other PRs may also solve one of the mentioned issues here, as they are not all uniform.

I am closing the issue now. If any such problem reappears, either comment here or open a new issue.
